File: First and last occurrences of X - GFG/first-and-last-occurrences-of-x.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    vector<int> firstAndLast(vector<int> &arr, int n, int x) {
        // Code here
        
        int s = 0,e = n-1;
        int f = -1,l = -1;
        for(int i = 0; i<arr.size(); i++){
            if(arr[i] == x){
                f = i;
                break;
            }
        }
         for(int i = arr.size()-1; i>=0; i--){
            if(arr[i] == x){
                l = i;
                break;
            }
        }
        
        if(f == -1 || l == -1) return {-1};
        return {f,l};
       
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, x;
        cin >> n >> x;
        vector<int> arr(n);
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }

        Solution obj;
        vector<int> ans= obj.firstAndLast(arr, n, x) ;
        for(int i:ans){
            cout<<i<<" ";
        }
        cout<< endl;
    }
    return 0;
}

// } Driver Code Ends

File: First and last occurrences of x - GFG/first-and-last-occurrences-of-x.java
//{ Driver Code Starts
//Initial Template for Java



import java.io.*;
import java.util.*;


// } Driver Code Ends
//User function Template for Java


class GFG
{   
    long binarySearch(long arr[],int n,int target,Boolean isMini){
        
        int start = 0;
        int end = n - 1;
        long ind = -1 ;
        while(start <= end){
            
            int mid = start + (end - start) /2 ;
            
            if(arr[mid] == target){
                ind = mid ;
                if(isMini) end = mid - 1;
                else start =  mid + 1;
            }else if(arr[mid] > target) end = mid - 1;
            else start = mid + 1;
        }
        
        return ind;
        
    }
    ArrayList<Long> find(long arr[], int n, int x)
    {
        // code here
        ArrayList<Long> ans = new ArrayList<Long>();
        
        if(n == 0) return ans;
        long mini = binarySearch(arr,n,x,true);
        long maxi = -1 ;
        
        if(mini != -1){
            maxi = binarySearch(arr,n,x,false);
        }
        
        ans.add(mini);
        ans.add(maxi);
        
        return ans;
    }
}



//{ Driver Code Starts.

// Driver class
class Array {

    // Driver code
    public static void main(String[] args) throws IOException {
        // Taking input using buffered reader
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int testcases = Integer.parseInt(br.readLine());
        // looping through all testcases
        while (testcases-- > 0) {
//            int n = Integer.parseInt(br.readLine());
            String line = br.readLine();
            String[] q = line.trim().split("\\s+");
            int n =Integer.parseInt(q[0]);
            int x =Integer.parseInt(q[1]);
//            //int y =Integer.parseInt(q[2]);
            String line1 = br.readLine();
            String[] a1 = line1.trim().split("\\s+");
            long arr[] = new long[n];
            for (int i = 0; i < n; i++) {
                arr[i] = Long.parseLong(a1[i]);
            }
            GFG ob = new GFG();
            ArrayList<Long> ans=ob.find(arr,n,x);
            System.out.println(ans.get(0)+" "+ans.get(1));
        }
    }
}

// } Driver Code Ends